src/goto-synthesizer/cegis_verifier.cpp

```diff
@@ -53,6 +53,37 @@ static bool contains_symbol_prefix(const exprt &expr, const std::string &prefix)
   return false;
 }
 
+static const exprt &
+get_checked_pointer_from_null_pointer_check(const exprt &violation)
+{
+  // A NULL-pointer check is the negation of an equation between the checked
+  // pointer and a NULL pointer.
+  // ! (POINTER_OBJECT(NULL) == POINTER_OBJECT(ptr))
+  const equal_exprt &equal_expr = to_equal_expr(to_not_expr(violation).op());
+
+  const exprt &lhs_pointer_object = equal_expr.lhs();
+  const exprt &rhs_pointer_object = equal_expr.rhs();
+  INVARIANT(
+    lhs_pointer_object.id() == ID_pointer_object &&
+      rhs_pointer_object.id() == ID_pointer_object,
+    "Null pointer check is an equation between two pointer objects of "
+    "pointer-type operands.");
+
+  const exprt &lhs_pointer = lhs_pointer_object.operands()[0];
+  const exprt &rhs_pointer = rhs_pointer_object.operands()[0];
+
+  // NULL == ptr
+  if(
+    can_cast_expr<constant_exprt>(lhs_pointer) &&
+    is_null_pointer(*expr_try_dynamic_cast<constant_exprt>(lhs_pointer)))
+  {
+    return rhs_pointer;
+  }
+
```

Comment:
Hmm, shouldn't you support NULL on either side of the equality? I'd expect an `else if(<same condition as above but using rhs_pointer>)`.

Reply:
No, the null-pointer check is
! (POINTER_OBJECT(null) == POINTER_OBJECT(ptr))
according to ```ansi-c/goto_check_c.cpp```.